### PR TITLE
std.range: Make OnlyResult define opEquals for convenience

### PR DESCRIPTION
**Pull request by ntrel**
`only(...)` is often used in unittests. Defining `OnlyResult.opEquals` helps to avoid importing `std.algorithm.comparison : equal`, and makes code easier to read (because it removes nesting of brackets).

* As `opEquals` is a template, `std.algorithm.comparison` is not imported unless `opEquals` is used.
* Encouraging use of `only` will help for `@nogc` code, although currently `std.algorithm.equal` doesn't work in `@nogc` code. This pull makes use of `only` more attractive vs array literals (which support `==` natively).

### CODE REVIEW

**Comment by ntrel** [comment]
Not sure whether to update the changelog (which says pending 2.072) or to file a bugzilla issue for this.

File: std/range/package.d [hunk 1]
```diff
@@ -7436,11 +7457,12 @@ auto only(Values...)(auto ref Values values)
 {
     import std.algorithm.comparison : equal;
     import std.algorithm.iteration : filter, joiner, map;
-    import std.algorithm.searching : findSplitBefore;
+    import std.algorithm.searching : find;
     import std.uni : isUpper;
 
-    assert(equal(only('♡'), "♡"));
-    assert([1, 2, 3, 4].findSplitBefore(only(3))[0] == [1, 2]);
```
Comment (ntrel) [inline std/range/package.d]: BTW This example is moving to: https://github.com/dlang/phobos/pull/4842

The `find` one is more general as it doesn't return a custom tuple.

**Comment by wilzbach** [comment]
> Not sure whether to update the changelog (which says pending 2.072) or to file a bugzilla issue for this.

I will update the changelog to files per entry tonight, but an issue works too.

**Comment by jmdavis** [comment]
Personally, I think that this is a bad idea. Ranges in general do not work with `==`, and I think that it's a bad idea to get in a habit of using any ranges that way. It's also going to result in confusion, because other ranges don't work this way.

**Pull request closed without merging**